cura/Arranging/GridArrange.py

```diff
@@ -242,13 +242,9 @@ def _moveNodeOnGrid(self, node: "SceneNode", grid_x: int, grid_y: int) -> "Opera
         center_grid_y = coord_grid_y + (0.5 * self._grid_height)
 
         bounding_box = node.getBoundingBox()
-        center_node_x = (bounding_box.left + bounding_box.right) * 0.5
-        center_node_y = (bounding_box.back + bounding_box.front) * 0.5
+        center_node_z = bounding_box.height / 2
 
-        delta_x = center_grid_x - center_node_x
-        delta_y = center_grid_y - center_node_y
-
-        return TranslateOperation(node, Vector(delta_x, 0, delta_y))
+        return TranslateOperation(node, Vector(center_grid_x, center_node_z, center_grid_y), True)
```

Comment:
Nice! didn't know we could set absolute coords; would have done this if I knew

One small nitpick, can we add a named argument for better code readability (so we know what the boolean refers to)?
```suggestion
        return TranslateOperation(node, Vector(center_grid_x, center_node_z, center_grid_y), set_position = True)
```